CRM/Contact/BAO/Query.php

```diff
@@ -514,7 +514,7 @@ public function initialize($apiEntity = NULL) {
       CRM_Financial_BAO_FinancialType::buildPermissionedClause($this->_whereClause, $component);
     }
 
-    $this->_fromClause = self::fromClause($this->_tables, NULL, NULL, $this->_primaryLocation, $this->_mode);
+    $this->_fromClause = self::fromClause($this->_tables, NULL, NULL, $this->_primaryLocation, $this->_mode, $apiEntity);
```

Comment:
Actually $this->_mode contain the constant values specified in https://github.com/jitendrapurohit/civicrm-core/blob/master/CRM/Contact/BAO/Query.php#L47  and ```apiEntity``` is assigned with string entity values e.g. Contact, Participant etc.